gateways/helpers/sendRequest/sendRequest.go

```diff
@@ -0,0 +1,58 @@
+package sendRequest
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"fmt"
+)
+
+type PayGoRequest interface {
+	GetUrl() (string, error)
+	GetJSON() ([]byte, error)
+	SetResponse([]byte) error
+}
+
+func SendJSONRequest(rq PayGoRequest, method string) error {
```

Comment:
Maybe you can split this method on two.
`prepareRequest`, `SendJSONRequest(rq PayGoRequest, method string)`

inside `SendJSONRequest` you can call `prepareRequest`.

Some gateways can be necessary call another content types, maybe we can create a method called `SendRequest` and pass a content type as parameter and inside this method we call a specific request.

What you think?